src/book/03-guides/01-main-codebase/02-maintaining-subsystems.mdx

```diff
@@ -0,0 +1,134 @@
+---
+section: Guides
+chapter: Main Codebase
+title: Maintaining Subsystems
+description: How to maintain subsystems within the main codebase.
+slug: /guides/main/maintaining-subsystems
+---
+
+This page details how to maintain various subsystems within the main codebase.
+
+# Vision
+
+Before vision can work on the robot, odometry and kinematics must work on the robot. If these are inaccurate, then the green horizon will be inaccurate.
+
+## Dataset Generation
+
+Datasets for training the Visual Mesh can be found on the Network-Attached Storage (NAS). If you don't have access to the NAS, ask a team leader for an account. The NAS contains millions of NUpbr training images and large TensorFlow records containing the correct data to train the Visual Mesh on.
+
+### NUpbr
+
+NUpbr is a Physically Based Rendering tool created in Blender. It creates semi-synthetic images with corresponding segmentation images for training.
+
+NUpbr requires Blender **2.79**. Download the appropriate file from the [Blender 2.79 Stable Release](https://download.blender.org/release/Blender2.79/).
+
+Clone the [NUpbr repository](https://github.com/NUbots/NUpbr) using
+
+```sh
+git clone https://github.com/NUbots/NUpbr.git
+```
+
+Navigate to the NUpbr directory and run
+
+```sh
+blender --python pbr.py
+```
+
+Find out more about NUpbr and how to edit the parameters on the [NUpbr NUbook page](/system/tools/nupbr).
+
+### NUgan
+
+The NUbots CycleGAN performs style transfer from the semi-synthetic style to a real image style. It creates more realistic images from the semi-synthetic images from NUpbr. It can be used for style transfer to a specific field for fine-tuning vision.
+
+Clone the [NUgan repository](https://github.com/NUbots/NUgan) using
+
+```sh
+git clone https://github.com/NUbots/NUgan.git
```

Comment:
It seems a bit strange to me to include one step of using NUgan, then send users to the repo docs anyway. I feel like you could remove this, and keep the rest, or add more steps as well.

Reply:
Yeah, it's currently redundant to have the link to the NUgan repo three times in three lines. It is somewhat convenient to have a copy paste link for cloning the repo though. One of the links to the github repo could probably be replaced with a link to a NUgan NUbook page.